internal/plan/plan.go

```diff
@@ -0,0 +1,73 @@
+package plan
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+)
+
+type PlanHandler struct {
+	OptimisationPlan OptimisationPlan
+}
+
+type OptimisationPlan struct {
+	SiteID                string                 `json:"site_id"`
+	OptimisationTimestamp OptimisationTimestamp  `json:"optimisation_timestamp"`
+	OptimisationIntervals []OptimisationInterval `json:"optimisation_intervals"`
+	SetpointType          int                    `json:"setpoint_type"`
+}
+
+type OptimisationTimestamp struct {
+	Seconds int64 `json:"seconds"`
+	Nanos   int64 `json:"nanos"`
+}
+
+type OptimisationInterval struct {
+	Interval      OptimisationIntervalTimestamp `json:"optimisation_interval"`
+	BatteryPower  OptimisationValue             `json:"battery_power"`
+	StateOfCharge float32                       `json:"state_of_charge"`
+	MeterPower    OptimisationValue             `json:"meter_power"`
+}
+
+type OptimisationIntervalTimestamp struct {
+	StartTime OptimisationTimestamp `json:"start_time"`
+	EndTime   OptimisationTimestamp `json:"end_time"`
+}
+
+type OptimisationValue struct {
+	Value float32 `json:"value"`
+	Unit  int     `json:"unit"`
+}
+
+func ReadPlan(path string) (OptimisationPlan, error) {
```

Comment:
question: If we are storing a single plan , which I believe, we are, what's the point of letting the client of this object to indicate where is the plan stored?
Shouldn't this abstraction encapsulate that information? so the client of this abstraction only cares about: Give me the latest plan, and also, asking this abstraction to just store the plan. But not caring about the _location_ of where the plan is stored.

Reply:
also, I would make both, `ReadPlan` and `StorePlan` part of the interface of the `PlanHandler` (methods)

Reply:
I guess if we're always running in a container then the client doesn't need to care where the file is stored ... it's just always been my habit to make any filesystem paths configurable, to avoid assumptions about where the service is running. But I guess that's not necessary in container world.

I did start out with PlanHandler methods but it seemed simpler this way... the names got confusing and there didn't seem much benefit to having to create a handler first. If the file location is non-configurable though, then I guess it makes more sense to have a handler which knows the location of the file and manages the read/write operations.

Reply:
@serroba I think a good compromise is to make the path to the plan a property of PlanHandler, which can optionally be set when the handler is initialised, but if not set will fall back to a default, and is not modifiable via config. That way tests can supply a different path, to ensure we avoid collisions, but the end user doesn't get to/have to specify the path. Does that sound ok?

Reply:
yes, that's what I had in mind.

Reply:
The path can be provided to the handler via its constructor. Is just, obscured from the client, but not from the initial configuration

Reply:
I've just been thinking though that we might want to write the file outside the container, so that if the container is recreated, the new one will pick up the existing file. If we're doing that, we'd need to specify the backup location as a volume in the docker-compose file and it would make more sense to provide the path as a configurable env var.

Reply:
Yes, all of that. But again, you can inject that information at the time of instantiation of the handler rather than from the client. The client of this shouldn't be caring about wherre the plan is stored. The handler is the one that should know where that is stored.

Reply:
ah got it. I was thinking of "the client" as the end user, i.e. whoever is writing the config file and setting up the container, not the code that's calling `ReadPlan`. So it's OK to have a config setting determining where the file goes, but that information should be provided to the handler when the handler is instantiated, not to the `ReadPlan` method when the file is being read.

Reply:
yes, that's exactly what I had in mind